### PR TITLE
feat: Enable ignoreResourceUpdate by default #19992

### PR DESCRIPTION
**Pull request by jaehanbyun**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [X] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

## What I have done:
*  Enable ignoreResourceUpdate by default

### CODE REVIEW

**Review by nitishfy — APPROVED** [review]
LGTM!

**Comment by crenshaw-dev** [comment]
Please also update documentation

**Comment by jaehanbyun** [comment]
> Please also update documentation
I updated documentation. Thanks!

File: docs/operator-manual/argocd-cm.yaml [hunk 1]
```diff
@@ -107,8 +107,8 @@ data:
     - /spec/replicas
 
   # Enable resource.customizations.ignoreResourceUpdates rules. If "false," those rules are not applied, and all updates
-  # to resources are applied to the cluster cache. Default is false.
-  resource.ignoreResourceUpdatesEnabled: "false"
+  # to resources are applied to the cluster cache. Default is true.
+  resource.ignoreResourceUpdatesEnabled: "true"
```
Comment (crenshaw-dev) [inline docs/operator-manual/argocd-cm.yaml:111]: Could you also update the docs in reconcile.md?

**Review by crenshaw-dev — DISMISSED** [review]

**Comment by jaehanbyun** [comment]
@crenshaw-dev I'm not sure if the option has been well explained, PTAL!

**Review by rumstead — APPROVED** [review]

**Review by christianh814 — APPROVED** [review]
Just needs to resolve feedback from Crenshaw but other than that LGTM

**Review by agaudreault — APPROVED** [review]
LGTM. We discussed it a bit today in the SIG Scalability. I have been involved in monitoring the usage of this feature through slack and GitHub issues, i haven't seen any negative impact after enabling it with the default configurations.

Obviously, it still needs to be configured to improve the performance significantly. Perhaps 3.0 can include default configuration for known resources (#19992) so i don't think it fully closes the issue.

Let's bring it up at the next contributors meeting just in case, but it's safe to merge after that IMO.

**Comment by jaehanbyun** [comment]
@agaudreault Understood. Please let me know if there are any additional details related to this PR after the meeting. Thanks!

**Pull request merged**